modules/boosted/tarot.service.ts

```diff
@@ -0,0 +1,105 @@
+import { GqlBalancePoolAprItem, GqlBalancePoolAprSubItem, GqlBalancerPool } from '../../schema';
+import { Cache, CacheClass } from 'memory-cache';
+import { getContractAt } from '../ethers/ethers';
+import TarotBoostedAbi from './abi/TarotBoostedAbi.json';
+import { tokenPriceService } from '../token-price/token-price.service';
+import { TokenPrices } from '../token-price/token-price-types';
+import _ from 'lodash';
+import { formatFixed } from '@ethersproject/bignumber';
+
+const TAROT_CACHE_KEY = 'tarot';
+
+interface TarotData {
+    address: string;
+    apr: number;
+}
+
+const TAROT_APR_CONTRACT_ADDRESS = '0xc32B9a98F2419544070E9580166C5A18Af8ADf9f';
+const tarotAPRContract = getContractAt(TAROT_APR_CONTRACT_ADDRESS, TarotBoostedAbi);
+
+export class TarotService {
+    private cache: CacheClass<string, TarotData[]>;
+    private boostedPools = [
+        {
+            address: '0xfB98B335551a418cD0737375a2ea0ded62Ea213b', // MAI
```

Comment:
don't hardcode these addresses like this.
- Use the factory address on the pool to determine its a tarot pool
- Use the `mainIndex` and `wrappedIndex` on the pool to determine main/wrapped tokens.

Reply:
I have hardcoded the factory address (for now). Is there a way to get it from the balancer sdk config? Do I need to resolve the 'network' key somehow first to get to the 'linearPools' key?

Reply:
@danielmkm I figured out how to grab the factory address from the balancer sdk so it is no longer hardcoded anymore.